### DISCUSSION TITLE
What is the difference between NotifyOfPropertyChange("") and NotifyOfPropertyChange(null)

### DISCUSSION THREAD

**Discussion by Kartoffelsalat2** [canton7/Stylet#382] [Q&A]
After calling either method, the getters of the properties bound in the view are called. Both refresh the view. Is there a difference?

**Comment by canton7** [comment]
Not for wpf. https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.inotifypropertychanged.propertychanged?view=net-7.0#remarks

**Comment by Kartoffelsalat2** [comment]
When i call NotifyOfPropertyChange(null) then the method private static IEventBinding BindImpl<TSource, TProperty>(TSource target, Expression<Func<TSource, TProperty>> targetSelector, EventHandler<PropertyChangedExtendedEventArgs<TProperty>> handler, bool invoke) where TSource : class, INotifyPropertyChanged in PropertyChangedExtensions.cs is called. In this method a handler is registered to the event ProperyChanged, but this handler considers only the case when the PropertyName is the empty string and not the case when the PropertyName is null. Which is why the inner handler is not called in the PropertyName == null case. Only unfortunately I don't know what the handler does.
